{"name": "test_allure2", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'int'", "trace": "self = <test_allure.TestAllure object at 0x000001C8DF6C8588>\n\n    @allure.step(title='allure1')\n    def test_allure2(self):\n>       allure.attach(2,\"输入2\")\n\nscripts\\test_allure.py:12: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\_allure.py:164: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_pytest\\listener.py:215: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\reporter.py:117: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <allure_commons.logger.AllureFileLogger object at 0x000001C8DF538E48>, body = 2, file_name = '55c54953-c6c7-45c6-9144-73f6512ec991-attachment.attach'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = os.path.join(self._report_dir, file_name)\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, text_type):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'int'\n\n..\\..\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\logger.py:62: TypeError"}, "steps": [{"name": "allure1", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'int'\n", "trace": "  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\_allure.py\", line 157, in impl\n    return func(*a, **kw)\n  File \"C:\\Users\\Administrator\\PycharmProjects\\allure_pratice\\scripts\\test_allure.py\", line 12, in test_allure2\n    allure.attach(2,\"输入2\")\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\_allure.py\", line 164, in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 87, in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_pytest\\listener.py\", line 215, in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\reporter.py\", line 117, in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 87, in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"c:\\users\\administrator\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\allure_commons\\logger.py\", line 62, in report_attached_data\n    attached_file.write(body)\n"}, "attachments": [{"name": "输入2", "source": "55c54953-c6c7-45c6-9144-73f6512ec991-attachment.attach"}], "start": 1589125535094, "stop": 1589125535095}], "start": 1589125535094, "stop": 1589125535095, "uuid": "fef6d1eb-f6ea-4976-b990-8a5b1bfbe970", "historyId": "d05ed3c3049a5242abbcf5674e7451c8", "testCaseId": "cbb8bebfe1a8faab30ab988f37193ab9", "fullName": "scripts.test_allure.TestAllure#test_allure2", "labels": [{"name": "parentSuite", "value": "scripts"}, {"name": "suite", "value": "test_allure"}, {"name": "subSuite", "value": "TestAllure"}, {"name": "host", "value": "WIN-DJ528GOHCBP"}, {"name": "thread", "value": "868-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "scripts.test_allure"}]}